pydarn/plotting/fan.py

```diff
@@ -197,38 +203,26 @@ def plot_fan(cls, dmap_data: List[dict], ax=None,
             # if there is no slist field this means partial record
             except KeyError:
                 continue
-
         # Begin plotting by iterating over ranges and beams
-        for gates in range(ranges[0], ranges[1] - 1):
-            for beams in range(thetas.shape[1] - 1):
-                # Index colour table correctly
-                cmapindex = (scan[gates, beams] + abs(zmin)) /\
-                        (abs(zmin) + abs(zmax))
-                if cmapindex < 0:
-                    cmapindex = 0
-
-                if cmapindex > 1:
-                    cmapindex = 1
-                colour_rgba = cmap(cmapindex)
-
-                # Check for zero values (white) and groundscatter (gray)
-                if scan[gates, beams] == 0:
-                    colour_rgba = (1, 1, 1, 0)
-
-                if groundscatter and grndsct[gates, beams] == 1:
-                    colour_rgba = 'gray'
-
-                # Angle for polar plotting
-                theta = [thetas[gates, beams], thetas[gates + 1, beams],
-                         thetas[gates + 1, beams + 1],
-                         thetas[gates, beams + 1]]
-                # Radius for polar plotting
-                r = [rs[gates, beams], rs[gates + 1, beams],
-                     rs[gates + 1, beams + 1], rs[gates, beams + 1]]
-                ax.fill(theta, r, color=colour_rgba)
+        thetas = thetas[ranges[0]:ranges[1]]
+        rs = rs[ranges[0]:ranges[1]]
+        scan = scan[ranges[0]:ranges[1]-1]
+        ax.pcolormesh(thetas, rs,
+                      np.ma.masked_array(scan, ~scan.astype(bool)),
+                      norm=norm, cmap=cmap)
+
+        # plot the groundscatter as grey fill
+        if groundscatter:
+            ax.pcolormesh(thetas, rs,
+                          np.ma.masked_array(grndsct,
+                                             ~grndsct.astype(bool)),
+                          norm=norm, cmap='Greys')
+
```

Comment:
Okay so the error is due to setting groundscatter to True, it works fine with it sent to False. So I'm assuming there's something wrong in these lines somewhere causing the above error.